homeassistant/components/nobo_hub/climate.py

```diff
@@ -0,0 +1,209 @@
+"""Python Control of Nobø Hub - Nobø Energy Control."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from pynobo import nobo
+
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    ATTR_TARGET_TEMP_HIGH,
+    ATTR_TARGET_TEMP_LOW,
+    PRESET_AWAY,
+    PRESET_COMFORT,
+    PRESET_ECO,
+    PRESET_NONE,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_IDENTIFIERS,
+    ATTR_MODE,
+    ATTR_NAME,
+    ATTR_SUGGESTED_AREA,
+    ATTR_VIA_DEVICE,
+    PRECISION_WHOLE,
+    TEMP_CELSIUS,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import (
+    ATTR_OVERRIDE_ALLOWED,
+    ATTR_SERIAL,
+    ATTR_TARGET_ID,
+    ATTR_TARGET_TYPE,
+    ATTR_TEMP_COMFORT_C,
+    ATTR_TEMP_ECO_C,
+    CONF_OVERRIDE_TYPE,
+    DOMAIN,
+    OVERRIDE_TYPE_NOW,
+)
+
+SUPPORT_FLAGS = (
+    ClimateEntityFeature.PRESET_MODE | ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+)
+
+PRESET_MODES = [PRESET_NONE, PRESET_COMFORT, PRESET_ECO, PRESET_AWAY]
+
+MIN_TEMPERATURE = 7
+MAX_TEMPERATURE = 40
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Nobø Ecohub platform from UI configuration."""
+
+    # Setup connection with hub
+    hub: nobo = hass.data[DOMAIN][config_entry.entry_id]
+
+    override_type = (
+        nobo.API.OVERRIDE_TYPE_NOW
+        if config_entry.options.get(CONF_OVERRIDE_TYPE) == OVERRIDE_TYPE_NOW
+        else nobo.API.OVERRIDE_TYPE_CONSTANT
+    )
+
+    # Add zones as entities
+    async_add_entities(
+        [NoboZone(zone_id, hub, override_type) for zone_id in hub.zones],
+        True,
+    )
+
+
+class NoboZone(ClimateEntity):
+    """Representation of a Nobø zone.
+
+    A Nobø zone consists of a group of physical devices that are
+    controlled as a unity.
+    """
+
+    _attr_max_temp = MAX_TEMPERATURE
+    _attr_min_temp = MIN_TEMPERATURE
+    _attr_precision = PRECISION_WHOLE
+    _attr_preset_modes = PRESET_MODES
+    # Need to poll to get preset change when in HVACMode.AUTO.
+    _attr_supported_features = SUPPORT_FLAGS
+    _attr_temperature_unit = TEMP_CELSIUS
+
+    def __init__(self, zone_id, hub: nobo, override_type):
+        """Initialize the climate device."""
+        self._id = zone_id
+        self._nobo = hub
+        self._attr_unique_id = f"{hub.hub_serial}:{zone_id}"
+        self._attr_name = hub.zones[self._id][ATTR_NAME]
+        self._attr_has_entity_name = False
```

Comment:
I had to set `_attr_has_entity_name` to False to avoid entity names like "Kitchen Kitchen" (if the zone is named "Kitchen"). However, should we add e.g. "Heater" as a suffix or prefix to the entity names?

Reply:
Changed after reading https://developers.home-assistant.io/blog/2022/07/10/entity_naming/  to 
```
self._attr_name = None
self._attr_has_entity_name = True
```
Seems to work as desired, but this is a bit confusing.

Reply:
Or is this better?
```
self._attr_name = "Heater"
self._attr_has_entity_name = True
```
Then I get entity names like `Kitchen Heater`

Other alternatives are `Heater(s)` or `Heater zone`

Reply:
If the entity represents the main feature of the device, the entity `_attr_name` should be set to `None` while `_attr_has_entity_name` is set to `True`.